### ISSUE TITLE
Ability to not document components vs ability to ignore components

### ISSUE THREAD

**Issue by AoDev** [sapegin/react-styleguidist#168]
Hello, 

I'd like to make a difference between component I want to ignore completely vs not documenting them. Some components are "helpers" or "private" components. I'd like to be able to use them in code examples without documenting them.

But, I still need styleguide to load them which is not possible with the current ignore option and so the application will throw an error.

Other use case and way to think about it: how can I make some components available in the whole documentation so that they can be used in code examples, without documenting them?

**Comment by sapegin** [comment]
There’s no ignore option (#11). What option do you mean?

**Comment by AoDev** [comment]
Right now it's possible to ignore them through the config.
```
    return glob
      .sync(path.resolve(FRAMEWORK_DIR, 'components', '**/*.js'))
      .filter(function shouldBeDocumented (module) {
        return (module.search(/someregex/) === -1)
      })
```

**Comment by sapegin** [comment]
Then two options:

1. Use `skipComponentsWithoutExample` instead of filtering components.
2. `require` modules you need in examples.

**Comment by sapegin** [comment]
Please check the docs updates in the previous commit: I’ve added some examples.

**Comment by AoDev** [comment]
Regarding point 2. I am not able to import modules in the examples. Maybe it works through modules inside node_modules that are resolved by webpack but if I want to import a "local" module then, how? 

Here is an example: in our docs, I've created a FakeBrowser component.

![screen shot 2016-08-18 at 11 51 44](https://cloud.githubusercontent.com/assets/1526150/17772465/582e6c12-6549-11e6-8e2f-1232b0aa0385.png)

The way I've found to make it work is to
- include it in the styleguide as a separate section.
- play with CSS so that the last section is hidden 

Like this:

section config (last section)

```
  {name: 'Docs components', components: function () {
    return glob
      .sync(path.resolve(FRAMEWORK_DIR, 'docs', 'components-for-docs', '**/*.js'))
      .filter(shouldBeDocumented)
  }}
```

```
// Hide the sections of components created for the docs
// We don't want people to see them
.ReactStyleguidist-TableOfContents__root >
.ReactStyleguidist-TableOfContents__list > li:last-child,
.ReactStyleguidist-Section__root:last-child {
  display: none;
}

.docs-fake-browser {
...
}
```

This is just an example. There are various different cases where I need components to be available to use in examples without them being documented, like private "helper components".
